model/Image.php

```diff
@@ -233,10 +247,10 @@ public function SetRatioSize($width, $height) {
 		$heightRatio = $height / $this->height;
 		if( $widthRatio < $heightRatio ) {
 			// Target is higher aspect ratio than image, so check width
-			if($this->isWidth($width) && !Config::inst()->get('Image', 'force_resample')) return $this;
+			if($this->isWidth($width)) return $this;
```

Comment:
There's an analogy in RESTful APIs that pass around URLs in href attributes from responses. Is the API allowed to change the URL form, and update the href attribute, and consider it a non-breaking change? That would probably come down to:
 
 * If the specific URL patterns were documented, then a change to the URL scheme would be breaking a public API, and wouldn't be appropriate without a major release bump.
 * If the URL patterns were not documented, and instead the API docs "make this request and then fetch an href from it", then a change to the URL scheme would be only breaking an internal API, and it probably can be done in a minor release bump.

So, it comes down to what's documented. Easy! 

Well, not so fast. In SilverStripe unfortunately we have a lot of things that are important not to break, and aren't necessarily documented. So saying "we might change anything not specifically spelled out in the documentation" is likely to ignore more people that is helpful.

Note that another way to deal with this would be negative documentation: something like "please call Link() methods and don't try and construct the URL yourself".

So, right now I don't think we're going to get unambiguous answers for this stuff, and we're going to need to make some assumptions over which undocumented APIs are still commonly-used enough to need to be treated as "public".

On the flipside, it would be great to push for more complete API documentation as we inch towards 4.0 (and, even better, tests that cover the same set of cases).

In the short term I would probably err on the side of conservatism for 3.x, since we don't know which features people are relying on.